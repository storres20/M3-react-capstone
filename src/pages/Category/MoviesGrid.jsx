import { useDispatch, useSelector } from 'react-redux';
import { useEffect, useState } from 'react';
import { useParams } from 'react-router-dom';
import MovieCard from '../../components/MovieCard/MovieCard';
import styles from './MoviesGrid.module.css';
import ScrollButton from '../../components/ScrollButton/ScrollButton';
import SearchBar from '../../components/SearchBar/SearchBar';
// Redux
import { getMovie } from '../../redux/movies/movies';
import { getAllData } from '../../redux/movies/allData';
// ***

export default function MoviesGrid() {
  const { catId } = useParams();

  /* useState - Loading */
  const [loading, setLoading] = useState(false);

  /* Setup Redux dispatch */
  const dispatch = useDispatch();

  /* Dispatch only once */
  useEffect(() => {
    dispatch(getMovie(catId));
    dispatch(getAllData(catId));
  }, [dispatch, catId]);

  /* Load Redux State */
  const movies = useSelector((state) => state.movie);
  const categories = useSelector((state) => state.category);
  // *********************************

  const category = categories.filter((item) => item.id.toString() === catId);

  useEffect(() => {
    setLoading(true);
  }, []);

  return (
    <div>
      {loading
        ? (
          <div>
            <h2 className={styles.title}>{category[0].name}</h2>

            {/* Filter Search Bar */}
            <SearchBar />

            <ul className={styles.moviesGrid}>
              {movies.map((movie) => <MovieCard key={movie.id} movie={movie} />)}
            </ul>

            {/* Scroll button for going back to TOP page */}
            <ScrollButton />
          </div>
        )

        : (
          <div className="flexLoad">
            <div className="lds-ring">
              <div />
              <div />
              <div />
              <div />
            </div>
          </div>
        )}

    </div>
  );
}
