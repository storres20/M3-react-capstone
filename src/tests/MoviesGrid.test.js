import React from 'react';
import renderer from 'react-test-renderer';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import MoviesGrid from '../pages/Category/MoviesGrid';
import moviesReducer from '../redux/movies/movies';
import categoriesReducer from '../redux/movies/categories';

// configureStore
const createTestStore = () => {
  const store = configureStore({
    reducer: {
      movie: moviesReducer,
      category: categoriesReducer,
    },
  });

  return store;
};
export default createTestStore;

// Test
describe('My Connected React-Redux Component', () => {
  let store;
  let component;

  beforeEach(() => {
    store = createTestStore();

    // Create a redux store
    component = renderer.create(
      <Provider store={store}>
        <MoviesGrid />
      </Provider>,
    ).toJSON();
  });

  test('renders correctly', () => {
    expect(component).toMatchSnapshot();
  });
});
