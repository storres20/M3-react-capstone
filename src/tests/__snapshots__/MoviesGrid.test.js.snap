// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Connected React-Redux Component renders correctly 1`] = `
<div>
  <div
    className="flexLoad"
  >
    <div
      className="lds-ring"
    >
      <div />
      <div />
      <div />
      <div />
    </div>
  </div>
</div>
`;
